notifications_utils/clients/redis/request_cache.py

```diff
@@ -1,17 +1,56 @@
 import json
+from collections import namedtuple
 from contextlib import suppress
 from datetime import timedelta
-from functools import wraps
+from functools import singledispatch, wraps
 from inspect import signature
 from uuid import UUID
 
 
 class RequestCache:
     DEFAULT_TTL = int(timedelta(days=28).total_seconds())
 
+    CacheResultWrapper = namedtuple("CacheResultWrapper", ("value", "cache_decision"))
+    CacheResultWrapper.__doc__ = """
+        Allows the result returned from a `RequestCache.set`-wrapped function to
```

Comment:
this docstring feels like implementation notes, rather than an actual docstring that describes what the thing is for, and how to use it. in particular, i think it's worth being explicit in the docstring that `value` is the value that will be cached and must be jsondumpable, and cache_decision is a bool

Reply:
I was mainly trying to ensure the behaviour of `cache_decision` was communicated to the potential user of the class.